pkg/endpoint/endpoint.go

```diff
@@ -1634,7 +1634,7 @@ func (e *Endpoint) BuilderSetStateLocked(toState, reason string) bool {
 	switch fromState { // From state
 	case StateCreating, StateWaitingForIdentity, StateReady, StateDisconnecting, StateDisconnected:
 		// No valid transitions for the builder
-	case StateWaitingToRegenerate:
+	case StateWaitingToRegenerate, StateRestoring:
```

Comment:
I'm trying to decide if the StateRestoring -> StateReady transition allowed here is ok. The comment suggests that it will be fine since it is only attempted after regeneration, so probably fine.